navigation/src/main/java/com/graphhopper/navigation/NavigateResource.java

```diff
@@ -188,13 +188,11 @@ private GHResponse calcRoute(List<Double> favoredHeadings, List<GHPoint> request
      * The url looks like: ".../{profile}/1.522438,42.504606;1.527209,42.504776;1.526113,42.505144;1.527218,42.50529?.."
      */
     private List<GHPoint> getPointsFromRequest(HttpServletRequest httpServletRequest, String profile) {
-
         String url = httpServletRequest.getRequestURI();
-        url = url.replaceFirst("/navigate/directions/v5/gh/" + profile + "/", "");
-        url = url.replaceAll("\\?[*]", "");
-
+        String urlStart = "/navigate/directions/v5/gh/" + profile + "/";
+        if (url.indexOf(urlStart) != 0) throw new IllegalArgumentException("Incorrect URL " + url);
```

Comment:
Maybe rather use `!url.startsWith("/navigate...")`?

Reply:
Ah, yeah, probably just `equals` is best.